R/showAreaAndBarPlots.R

```diff
@@ -161,24 +163,30 @@ showAreaAndBarPlots <- function(
         mapping = aes(.data$period, .data$value),
         size = 1.3
       )
-    dRegiTot <- dnohist %>%
-      filter(
-        .data$region != .env$mainReg,
-        .data$variable == .env$tot) %>%
-      droplevels()
-    dRegiTot$scenario <- dRegiTot$identifier
-    p4 <- p4 +
-      geom_line(
-        data = dRegiTot,
-        mapping = aes(.data$period, .data$value),
-        size = 1.3
-      )
+    if (length(unique(data$region)) > 1) {
+      dRegiTot <- dnohist %>%
+        filter(
+          .data$region != .env$mainReg,
+          .data$variable == .env$tot) %>%
+        droplevels()
+      dRegiTot$scenario <- dRegiTot$identifier
+      p4 <- p4 +
+        geom_line(
+          data = dRegiTot,
+          mapping = aes(.data$period, .data$value),
+          size = 1.3
+        )
+    }
   }
 
   # Show plots.
-  grid.arrange(p1, p2, p3, layout_matrix = rbind(c(1, 3), c(2, 3)), left = label)
-  cat("\n\n")
-  print(p4)
+  if (length(unique(data$region)) > 1) {
+    grid.arrange(p1, p2, p3, layout_matrix = rbind(c(1, 3), c(2, 3)), left = label)
+    cat("\n\n")
+    print(p4)
+  } else {
+    grid.arrange(p1, p2, layout_matrix = rbind(c(1, 2)), left = label)
```

Comment:
- Do I understand that correctly that you place the two plot next to each other?
- An alternative would be to define `p3 <- grid.rect(gp=gpar(col="white"))` or something like that and have the same formatting of the first page, but with a blank on the left side. I leave that up to you.

Reply:
> Do I understand that correctly that you place the two plot next to each other?

Yes, this is the layout when there is a single region in the data:

![image](https://github.com/pik-piam/mip/assets/33692508/8687ab47-b722-44c6-a9e6-d4be33e76658)